### DISCUSSION TITLE
Can I view factory method generated assets on UI?

### DISCUSSION THREAD

**Discussion by stevenmurphy12** [dagster-io/dagster#18540] [Q&A]
I'm generating an asset using a factory pattern, inspired from here - https://dagster.io/blog/python-factory-patterns

`_my_asset` contains a lot of code that I'd like to re-use across many assets, and parameterise on a case-by-case basis as required.

Right now I pass in my params to the generator method which creates the asset. Is it possible I could do this with a configuration object, then have that configuration appear in the user interface? In the example below, the properties of `SomeConfigClass` appear on the UI, and can be edited via the launchpad, I'd like to be able to do the same with my factory paramters.

```python
the_asset = generate_my_asset(param_1="aaa", param_2="bbb")

def generate_my_asset(param_1, param_2):
    
    @asset
    def _my_asset(context, config: SomeConfigClass):
        print(param_1)
        print(param_2)
        #Do Stuff
        
    return _my_asset
```

**Comment by yuhan** [comment]
We don't have the native UI support for this. But I guess you could construct a yaml for this pattern. Here's an example: https://github.com/dagster-io/dagster/tree/master/examples/experimental/assets_yaml_dsl/assets_yaml_dsl/pure_assets_dsl

**Reply by geoHeil** [reply]
Take a look at https://github.com/dagster-io/dagster/discussions/18568#discussioncomment-7792734 maybe it is helfpufl

**Reply by geoHeil** [reply]
I.e. not sure if I understand you correctly - but if you do NOT need the inheritance which was important for me then the example from the link should probably work.

**Reply by stevenmurphy12** [reply]
> We don't have the native UI support for this.

Appreciate the info. Whilst the YAML DSL doesn't handle this particular question, it's given me food for thought in other areas thank you

**Reply by stevenmurphy12** [reply]
I'm now in an odd position by which something works but I don't know why:

```python
the_asset = generate_my_asset(param_1="aaa", param_2="bbb")
the_asset_2 = generate_my_asset(param_1="ccc", param_2="ddd")

def generate_my_asset(param_1, param_2):
    
    class SomeConfigClass(Config):
       param_a: str = param_1
       param_b: str = param_2

    @asset
    def _my_asset(context, config: SomeConfigClass):
        print(config.param_a)
        print(config.param_b)
        #Do Stuff
        
    return _my_asset
```

I've changed it to embed the config class in the asset factory method. The values now appear on UI as expected. However, I didn't think this would work, how would the asset know what instance of `SomeConfigClass` to use? (if I define two assets using this factory).

Only just got this working though, will report back if I see any weirdness
